### PR TITLE
Fix String#camelcase(lower: true)

### PR DESCRIPTION
**Pull request by asterite**
Correctly implements #7717

### CODE REVIEW

**Comment by vladfaust** [comment]
I'm sorry, but `camelcase(downcase: true)` is tautology. It's even worse given the fact that we have `downcase` method.

**Comment by asterite** [comment]
The thing is that programming will never read as a natural language so from time to time you have to make compromises. It's also not a big deal, this is not a method that's going to be used a lot.

**Comment by asterite** [comment]
The thing is that programming will never read as a natural language so from time to time you have to make compromises. It's also not a big deal, this is not a method that's going to be used a lot.

**Comment by bew** [comment]
What about `camelcase(first_low: true)` ?

**Comment by asterite** [comment]
Maybe we can just have `lower_camelcase` and that's it.

**Comment by j8r** [comment]
@asterite @vladfaust ideally, we have to express:
- camelcase - already in the method name (nothing to do)
- lower/down case - `down` or `lower`
- the first character is impacted - `first`

So we either have `down_first` or `lower_first`, as @straight-shoota suggested before.
`first_down` and `first_lower` are also solutions, but less readable.

File: src/string.cr [hunk 1]
```diff
@@ -3499,21 +3499,23 @@ class String
 
   # Converts underscores to camelcase boundaries.
   #
+  # If *lower* is true, lower camelcase will be returned (the first letter is downcased).
+  #
   # ```
   # "eiffel_tower".camelcase                                            # => "EiffelTower"
   # "empire_state_building".camelcase(downcase: true)                   # => "empireStateBuilding"
```
Comment (bew) [inline src/string.cr]:
```suggestion
  # "empire_state_building".camelcase(lower: true)                      # => "empireStateBuilding"
```

**Comment by straight-shoota** [comment]
From https://github.com/crystal-lang/crystal/pull/7731#issuecomment-488674607 by @asterite 

> Actually, I think `lower` is fine. In wikipedia there's "upper camel case" and "lower camel case", so `camelcase(lower: true)` would use the "lower" variant.

+1 from me

**Comment by asterite** [comment]
@vladfaust Thank you for the suggestion, I applied the changes.

File: src/string.cr (hunk 1 above)
Comment (straight-shoota) [reply src/string.cr]: @asterite This needs to be changed

Comment (asterite) [reply src/string.cr]: Thanks

**Pull request merged**